device/celestica/x86_64-cel_silverstone-r0/plugins/sfputil.py

```diff
@@ -344,11 +393,146 @@ def reset(self, port_num):
 
         return True
 
+    def _page_to_flat(self, addr, page = -1):
+        flat = 0
+        if addr < 128:
+            flat = addr
+        else:
+            flat = ((page + 1) << 7) | (addr & 0x7f)
+        return flat
+
+    def _write_byte(self, port_num, devid, page, off, val):
+        eeprom_path = self._get_port_eeprom_path(port_num, devid)
+        addr = self._page_to_flat(off, page)
+        try:
+            f = open(eeprom_path, "wb", 0)
+            f.seek(addr)
+            f.write(chr(val))
+        except Exception as ex:
+            log_info("write failed: {0}".format(ex))
+        finally:
+            f.close()
+
+    def _init_cmis_module(self, port_num):
+        buf = self._read_eeprom_devid(port_num, self.IDENTITY_EEPROM_ADDR, 0x80, 64)
+        if buf is None:
+            log_info("PORT {0}: unable to read PAGE0".format(port_num))
+            return False
+        # Skip, in case of QSFP28
+        if buf[0] not in "18,19":
+            log_info("PORT {0}: skipped, it's not a QSFPDD".format(port_num))
+            return True
+
+        # Decode the transceiver ids
+        name = self.inf8628.parse_vendor_name(buf, 1)['data']['Vendor Name']['value']
+        part = self.inf8628.parse_vendor_pn(buf, 20)['data']['Vendor PN']['value']
+        #log_info("_init_cmis_module: p={0}, id='{1}', name='{2}', part='{3}'".format(port_num, buf[0], name, part))
+        # As of now, init sequence is only necessary for 'InnoLight T-DP4CNT-N00'
+        if name.upper() != 'INNOLIGHT' or part.upper() != 'T-DP4CNT-N00':
+            return True
+
+        log_info("PORT {0}: _init_cmis_module".format(port_num))
+
+        # Allow 1s for software reset
+        self._write_byte(port_num, self.IDENTITY_EEPROM_ADDR, -1, 26, 0x08)
```

Comment:
it would be good to wrap below logic into a function to have hwsku, module name as input so easier to expand?  Also, are we deprecating the cmis-init.sh files, if so, remove both of them from the PR.

Reply:
1. Ack, will do
2. Yes, the use of cmis-init.sh is deprecated, it's a known issue the InnoLight T-DP4CNT-N00 could sometimes failed and we'll need to poll its state for the retry. And the script that we provided is also for concept approval, it's not expected to have it directly integrated as-is.

Reply:
can you remove both cmis-init.sh from the PR?

Reply:
Done